examples/all-clusters-app/all-clusters-common/src/EnergyEvseDelegateImpl.cpp

```diff
@@ -0,0 +1,771 @@
+
+    /* update MinimumChargeCurrent & MaximumChargeCurrent to 0 */
+    SetMinimumChargeCurrent(0);
+    SetMaximumChargeCurrent(0);
+
+    /* update MaximumDischargeCurrent to 0 */
+    SetMaximumDischargeCurrent(0);
+
+    // TODO: Generate events
+
+    return Status::Success;
+}
+
+/**
+ * @brief   Called when EVSE cluster receives EnableCharging command
+ *
+ * @param chargingEnabledUntil (can be null to indefinite charging)
+ * @param minimumChargeCurrent (in mA)
+ * @param maximumChargeCurrent (in mA)
+ */
+Status EnergyEvseDelegate::EnableCharging(const DataModel::Nullable<uint32_t> & chargingEnabledUntil,
+                                          const int64_t & minimumChargeCurrent, const int64_t & maximumChargeCurrent)
+{
+    ChipLogProgress(AppServer, "EnergyEvseDelegate::EnableCharging()");
+
+    if (maximumChargeCurrent < kMinimumChargeCurrent || maximumChargeCurrent > kMaximumChargeCurrent)
+    {
+        ChipLogError(NotSpecified, "Maximum Current outside limits");
+        return Status::ConstraintError;
+    }
+
+    if (minimumChargeCurrent < kMinimumChargeCurrent || minimumChargeCurrent > kMaximumChargeCurrent)
+    {
+        ChipLogError(NotSpecified, "Maximum Current outside limits");
+        return Status::ConstraintError;
+    }
+
+    if (minimumChargeCurrent > maximumChargeCurrent)
+    {
+        ChipLogError(NotSpecified, "Minium Current > Maximum Current!");
+        return Status::ConstraintError;
+    }
+
+    if (chargingEnabledUntil.IsNull())
+    {
+        /* Charging enabled indefinitely */
+        ChipLogError(NotSpecified, "Charging enabled indefinitely");
+    }
+    else
+    {
+        /* check chargingEnabledUntil is in the future */
+        ChipLogError(NotSpecified, "Charging enabled until: %lu", static_cast<long unsigned int>(chargingEnabledUntil.Value()));
+        // TODO
+        // if (checkChargingEnabled)
+    }
+
+    /* Check current state isn't already enabled */
+
+    /* If charging is already enabled, check that the parameters may have
+       changed, these may override an existing charging command */
+    switch (mHwState)
+    {
+    case StateEnum::kNotPluggedIn:
+        SetState(StateEnum::kNotPluggedIn);
+        break;
+
+    case StateEnum::kPluggedInNoDemand:
+        SetState(StateEnum::kPluggedInNoDemand);
+        break;
+
+    case StateEnum::kPluggedInDemand:
+        /* If the EVSE is asking for demand then enable charging */
+        SetState(StateEnum::kPluggedInCharging);
+        break;
+
+    default:
+        ChipLogError(AppServer, "Unexpected EVSE hardware state\n");
+        SetState(StateEnum::kFault);
+        break;
+    }
+
+    /* update SupplyState */
+    SetSupplyState(SupplyStateEnum::kChargingEnabled);
+
+    /* If it looks ok, store the min & max charging current */
+    mMaximumChargingCurrentLimitFromCommand = maximumChargeCurrent;
+    SetMinimumChargeCurrent(minimumChargeCurrent);
+    // TODO persist these to KVS
```

Comment:
Could we create an issue with checkmarks and reference it here? I.e. who does these todos and when.

Reply:
https://github.com/project-chip/connectedhomeip/issues/30997